src/modules/trade/actions/update-trades-in-progress.js

```diff
@@ -91,51 +92,61 @@ export function updateTradesInProgress(marketID, outcomeID, side, numShares, lim
       cleanLimitPrice = outcomeTradeInProgress.limitPrice
     }
 
-    if (cleanNumShares && !cleanLimitPrice && (market.type === SCALAR || cleanLimitPrice !== '0')) {
+    if (cleanNumShares && !cleanLimitPrice && (market.marketType === SCALAR || cleanLimitPrice !== '0')) {
       cleanLimitPrice = topOrderPrice
     }
-
     // if this isn't a scalar market, limitPrice must be positive.
-    if (market.type !== SCALAR && limitPrice) {
+    if (market.marketType !== SCALAR && limitPrice) {
       cleanLimitPrice = bignumLimit.abs().toFixed() || outcomeTradeInProgress.limitPrice || topOrderPrice
     }
 
+    // if market order limitPrice should be null
+    if (orderType === MARKET) {
+      cleanLimitPrice = null
+    }
+
     const newTradeDetails = {
       side: cleanSide,
       numShares: cleanNumShares === '0' ? undefined : cleanNumShares,
       limitPrice: cleanLimitPrice,
       totalFee: '0',
       totalCost: '0'
     }
-
     // trade actions
-    if (newTradeDetails.side && newTradeDetails.numShares && loginAccount.address) {
+    if (newTradeDetails.side && loginAccount.address && newTradeDetails.numShares) {
       dispatch(loadAccountPositions({ market: marketID }, (err, accountPositions) => {
         if (err) {
           return dispatch({
             type: UPDATE_TRADE_IN_PROGRESS,
             data: { marketID, outcomeID, details: newTradeDetails }
           })
         }
+        const cleanAccountPositions = []
+        for (let i = 0; i < market.numOutcomes; i++) {
+          if (accountPositions[i] && !isNaN(accountPositions[i].numShares)) {
+            cleanAccountPositions.push(accountPositions[i].numShares)
+          } else {
+            cleanAccountPositions.push(0)
```

Comment:
Should this be a string `'0'` instead of JS number `0`?  In `newTradeDetails`, num shares appears to be a string.  What type is `accountPositions[i].numShares` expected to be?  (String, JS number, BigNumber, something else?)

Either way, in general, as mentioned above: avoid JS numbers.